### PR TITLE
build: deploy screenshot functions

### PR DESCRIPTION
**Pull request by DevVersion**
* Deploys the screenshot functions for push builds
* Runs deploying & publishing in parallel

### CODE REVIEW

File: scripts/release/deploy-screenshot-functions.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Go to the project root directory
+cd $(dirname ${0})/../..
+
+# Install node modules for screenshot functions. Firebase CLI needs to execute the functions
+# before it can collect all functions and deploy them.
+(cd tools/screenshot-test/functions; npm install)
+
+if [ -z ${MATERIAL2_SCREENSHOT_ACCESS_TOKEN} ]; then
```
Comment (jelbourn) [inline scripts/release/deploy-screenshot-functions.sh]: Created `MATERIAL2_SCREENSHOT_FIREBASE_DEPLOY_TOKEN` for this

File: scripts/release/publish-build-artifacts.sh [hunk 2]
```diff
@@ -30,8 +33,10 @@ publishPackage() {
   repoUrl="https://github.com/angular/${packageRepo}.git"
   repoDir="tmp/${packageRepo}"
 
-  # Create a release of the current repository.
-  $(npm bin)/gulp ${packageName}:build-release:clean
+  if [[ ! ${COMMAND_ARGS} == *--no-build* ]]; then
```
Comment (jelbourn) [inline scripts/release/publish-build-artifacts.sh]: What is this flag needed for?

File: scripts/release/deploy-screenshot-functions.sh (hunk 1 above)
Comment (DevVersion) [inline scripts/release/deploy-screenshot-functions.sh]: Thanks!

File: scripts/release/publish-build-artifacts.sh (hunk 2 above)
Comment (DevVersion) [inline scripts/release/publish-build-artifacts.sh]: This allows us to decide whether the script itself should build or if we want to build the output before on our own.

This allows us to run publishing & deploying in parallel.

**Pull request merged**